app/src/main/java/com/goldenraven/padawanwallet/ui/drawer/SendCoinsBackScreen.kt

```diff
@@ -1,15 +1,107 @@
+import androidx.compose.ui.text.PlaceholderVerticalAlign
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.navigation.NavController
+import com.goldenraven.padawanwallet.R
 import com.goldenraven.padawanwallet.ui.DrawerAppBar
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
 
 
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 internal fun SendCoinsBackScreen(navController: NavController) {
+
+    val context = LocalContext.current
+    val scope = rememberCoroutineScope()
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    val copyAddrString = buildAnnotatedString {
+        appendInlineContent(id = "copyAddrImageId")
+        append(stringResource(id = R.string.copyAddrStr))
+    }
+
+    val inlineContentMap = mapOf(
+        "copyAddrImageId" to InlineTextContent(
+            Placeholder(17.sp, 17.sp, PlaceholderVerticalAlign.TextCenter)
+        ) {
+            Image(
+                painterResource(R.drawable.ic_copy_outline),
+                contentDescription = "Copy to clipboard image",
+            )
+        }
+    )
+
     Scaffold(
         topBar = { DrawerAppBar(navController, title = "Send Coins Back") },
-    ) { }
+    ) {
+        Column(
+            Modifier.padding(all = 32.dp)
+        ) {
+            Image(
+                painterResource(R.drawable.return_sats_faucet_address),
+                contentDescription = "Return sats faucet address image",
+                modifier = Modifier.padding(start = 50.dp, end = 50.dp, bottom = 20.dp)
+            )
+            Column(
+                Modifier
+                    .align(Alignment.CenterHorizontally)
+                    .clickable(onClick = { copyToClipboard(context, scope, snackbarHostState) })
+                    .padding(bottom = 20.dp)
+            ) {
+                Text(
+                    text = stringResource(id = R.string.send_coins_back_address),
+                    maxLines = 1
+                )
+                Divider(
+                    color = colorResource(id = R.color.fg1),
+                    thickness = 1.dp,
+                    modifier = Modifier.padding(all = 3.dp)
+                )
+                Text(
+                    text = copyAddrString,
+                    inlineContent = inlineContentMap,
+                    modifier = Modifier.align(Alignment.End)
+                )
+            }
+            Text(text = stringResource(id = R.string.send_coins_back))
+            SnackbarHost(hostState = snackbarHostState)
+        }
+    }
 }
+
+fun copyToClipboard(context: Context, scope: CoroutineScope, snackbarHostState: SnackbarHostState) {
+    val clipboard: ClipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+    val clip: ClipData = ClipData.newPlainText("Copied address", context.getString(R.string.send_coins_back_address))
```

Comment:
On my device the end of the address is not shown, but there is also no real indication that a part of it is missing. It's not a big deal because you click/copy anyway the right string, but I'm wondering if you'd mind making the font a bit smaller so that it shows the full address on more devices.

Reply:
Maybe not a long-term solution, but for now at least is a bit better than just cutting the address on most devices.